lib/core/utils/qsa.js

```diff
@@ -44,10 +44,10 @@ function matchSelector (targets, exp, recurse) {
 	targets = Array.isArray(targets) ? targets : [targets];
 	targets.forEach((target) => {
 		if (matchesTag(target.actualNode, exp) &&
-			matchesClasses(target.actualNode, exp) &&
-			matchesAttributes(target.actualNode, exp) &&
-			matchesId(target.actualNode, exp) &&
-			matchesPseudos(target, exp)) {
+				matchesClasses(target.actualNode, exp) &&
```

Comment:
These are refactors that should not be included in a fix commit

Reply:
Can you explain what the value is in doing things like that? We need a stricter commit policy if that's what you're looking for in it. I was going to squash this PR into a single commit. I've done that for most PRs, except if there were clearly multiple features / fixes in them. I was under the impression that that was the preferred approach.